src/core/mac/mac.cpp

```diff
@@ -1587,8 +1594,27 @@ Error Mac::ProcessReceiveSecurity(RxFrame &aFrame, const Address &aSrcAddr, Neig
         break;
 
     case Frame::kKeyIdMode2:
-        macKey     = &mMode2KeyMaterial;
-        extAddress = &AsCoreType(&sMode2ExtAddress);
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+        if (aFrame.IsWakeupFrame())
+        {
+            uint32_t sequence;
+
+            // TODO: Avoid generating a new key if a wake-up frame was recently received already
+
+            IgnoreError(aFrame.GetKeyId(keyid));
+            sequence = BigEndian::ReadUint32(aFrame.GetKeySource());
```

Comment:
Just double-checking the `BigEndian` encoding for the sequence number (part of the key source data). Is this intentional, since most `Mac::Frame` fields are little-endian?

Reply:
TBH I don't know what led to use this endianess (maybe inspired by some usage of big endianess for key sequence in `KeyManager`), but this is how the devices using the prototype firmware behave now, so I'd suggest to keep it as is for now to avoid backwards incompatibilities. Also the spec is not totally clear on whether the Key Source field is a number or a string.

We can always keep track of this issue in Thread Group to take a final decision before going to production/certification.

Reply:
The Key Source field is defined as an octet string.

See [SPEC-179](https://threadgroup.atlassian.net/browse/SPEC-179) for prior discussion on this.

Reply:
Using big-endian here would be consistent with:

https://github.com/openthread/openthread/blob/213665cce005b88097c3a9ccace9a52767f551fe/src/core/thread/mle.hpp#L1133

Should we revert to big-endian?

Reply:
Thanks. Keeping `sequence = BigEndian::ReadUint32(aFrame.GetKeySource())` seems reasonable, especially if MLE uses the same approach.